src/status_im/ui2/screens/common/alert/view.cljs

```diff
@@ -0,0 +1,59 @@
+(ns status-im.ui2.screens.common.alert.view
+  (:require
+   [reagent.core :as reagent]
+   [status-im.i18n.i18n :as i18n]
+   [status-im.utils.re-frame :as rf]
+   [quo2.components.markdown.text :as quo2.text]
+   [react-native.core :as rn]
+   [quo2.core :as quo2]
+   [quo2.components.buttons.button :as quo2.button]
+   [status-im.switcher.constants :as switcher.constants]
+   [status-im.ui2.screens.common.alert.style :as style]
+   [quo2.components.selectors.selectors :as selectors]))
+
+(defn avatar [group-chat color display-name photo-path]
+  (if group-chat
+    [quo2/group-avatar {:color color
+                        :size  :small}]
+    [quo2/user-avatar {:full-name        display-name
+                       :profile-picture  photo-path
+                       :size             :xxs
+                       :status-indicator false}]))
+
+(defn extra-action-view [extra-action extra-text extra-action-selected?]
+  (when extra-action
+    [rn/view {:style {:margin-top 16 :flex-direction :row}}
+     [selectors/checkbox {:on-change (fn [^js e] (reset! extra-action-selected? e))}]
+     [quo2.text/text {:style {:margin-left 10}} extra-text]]))
+
+(defn alert [{:keys [title description context button-text on-press extra-action extra-text]}]
+  (let [extra-action-selected? (reagent/atom false)]
+    (fn []
+      (let [{:keys [group-chat chat-id public-key color name]} context
+            id           (or chat-id public-key)
+            display-name (if-not group-chat (first (rf/sub [:contacts/contact-two-names-by-identity id])) name)
+            contact      (when-not group-chat (rf/sub [:contacts/contact-by-address id]))
+            photo-path   (when-not (empty? (:images contact)) (rf/sub [:chats/photo-path id]))
+            {:keys [width]} (switcher.constants/dimensions)
```

Comment:
here 2 different issues:
1) for sure `switcher.constants` can't be used here we should take dimensions from react-native
2) we should avoid using width, flex-box should be used instead, align center, margins etc , not magic like `(/ (- width 52) 2)`

Reply:
Thank you! fixed it.